sklearn/neighbors/tests/test_nca.py

```diff
@@ -109,9 +109,9 @@ def fun(M):
     def grad(M):
         return nca._loss_grad_lbfgs(M, X, mask)[1]
 
-    # compute relative error
-    rel_diff = check_grad(fun, grad, M.ravel()) / np.linalg.norm(grad(M))
-    np.testing.assert_almost_equal(rel_diff, 0.0, decimal=5)
+    # compare the gradient to a finite difference approximation
+    diff = check_grad(fun, grad, M.ravel())
+    assert diff == pytest.approx(0.0)
```

Comment:
Note: testing the relative difference was not really meaningful because the gradient can be arbitrary close to 0 depending on the data.